lib/matplotlib/figunits.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import six
+import warnings
+import numpy as np
+from matplotlib.layoutbox import get_renderer
+
+"""
+Simple utility functions to parse and translate common
+drawing units
+"""
+
+suffixes = ['in', 'cm', 'mm', 'px', 'pt', 'em']
+cmperin = 2.54
+mmperin = 25.4
+ptperin = 72.272
+
+
+def tofigw(lenstr, fig):
+    """
+
+    """
+    return tofig(lenstr, fig, 'width')
+
+
+def tofigh(lenstr, fig):
+    """
+
+    """
+    return tofig(lenstr, fig, 'height')
+
+
+def tofig(lenstr, fig, dir):
+    """
+
+    """
+
+    if type(lenstr) == int or type(lenstr) == float:
```

Comment:
isinstance(lenstr, numbers.Number) / is_numlike (which does the same).  This allows also supporting numpy scalars.